### PR TITLE
Add differential privacy functions

### LINKED ISSUE #6874: Integrate differential privacy algorithms
https://github.com/google/differential-privacy/

Need to check the API of this library. Is it necessarily to collect values in array before calculation? How it will be possible to merge intermediate aggregation states? It's always possible to use array of all values as an intermediate state but it should be considered as worst case scenario.

### PR DESCRIPTION
**Pull request by tafxle**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- New Feature

Changelog entry:
Added differentially private aggregation functions 

resolves #6874

### CODE REVIEW

**Comment by nikitamikhaylov** [comment]
I think the issue was to import that library to our `/contrib`, but I don't see it in this PR. In addition, I don't see any test.
According to [this](https://github.com/frankmcsherry/blog/blob/master/posts/2016-02-06.md) and [this](https://arxiv.org/pdf/1909.01917.pdf) papers, we can implement generating of Laplace distribution and computing Pearson correlation coefficient in our codebase.. But it is unnecessary due to external library existence. 
Maybe, I don't understand your idea, so tell me more about it and write comments in code )

**Comment by alexey-milovidov** [comment]
If I remember correctly, Artemy has found out that the library is not directly applicable.

File: src/AggregateFunctions/AggregateFunctionDifferentiallyPrivate.h [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+        static_cast<ColVecType &>(to).getData().push_back(right);
+    }
+
+private:
+    Float32 level;
+};
+
+/**
+ * @tparam Value - type of values
+ * @tparam binary_bins - use bins of power of 2
```
Comment (nikitamikhaylov) [inline src/AggregateFunctions/AggregateFunctionDifferentiallyPrivate.h:476]: Remove unused parameters from comment, because is causes build fails.

**Comment by nikitamikhaylov** [comment]
This feature has unexpected results: (with big epsilon functions with `anon_` prefix must be equal to ordinary function without prefix)
```sql
jakalletti.N1 :) select anon_sum(100000000)(number) from numbers(10)

SELECT anon_sum(100000000)(number)
FROM numbers(10)

┌─anon_sum(100000000)(number)─┐
│                           9 │
└─────────────────────────────┘

1 rows in set. Elapsed: 0.039 sec. 

jakalletti.N1 :) select anon_max(100000000)(number) from numbers(10)

SELECT anon_max(100000000)(number)
FROM numbers(10)

┌─anon_max(100000000)(number)─┐
│          203886465925189687 │
└─────────────────────────────┘

1 rows in set. Elapsed: 0.007 sec. 

jakalletti.N1 :) select anon_max(100000000)(number) from numbers(100000)

SELECT anon_max(100000000)(number)
FROM numbers(100000)

┌─anon_max(100000000)(number)─┐
│          782404637512773607 │
└─────────────────────────────┘
```
And even on test dataset.
```sql
jakalletti.N1 :) select UserID, anon_max(100000000)(CounterID) from datasets.hits_v1 group by UserID order by UserID limit 10

SELECT 
    UserID,
    anon_max(100000000)(CounterID)
FROM datasets.hits_v1
GROUP BY UserID
ORDER BY UserID ASC
LIMIT 10

┌──────────UserID─┬─anon_max(100000000)(CounterID)─┐
│   2961521519262 │                     1289014762 │
│  87878526839192 │                       77389592 │
│ 143796275655540 │                      306508325 │
│ 160547676589519 │                      631915114 │
│ 164599821266083 │                      992350097 │
│ 172811142541414 │                     1909887272 │
│ 183173490300605 │                       57724458 │
│ 185609125465943 │                      979153435 │
│ 189975612000676 │                     1470608026 │
│ 190487283562760 │                      373731947 │
└─────────────────┴────────────────────────────────┘

jakalletti.N1 :) select UserID, max(CounterID) from datasets.hits_v1 group by UserID order by UserID limit 10

SELECT 
    UserID,
    max(CounterID)
FROM datasets.hits_v1
GROUP BY UserID
ORDER BY UserID ASC
LIMIT 10

┌──────────UserID─┬─max(CounterID)─┐
│   2961521519262 │       30847342 │
│  87878526839192 │       33265442 │
│ 143796275655540 │       16015323 │
│ 160547676589519 │       14625884 │
│ 164599821266083 │       27870005 │
│ 172811142541414 │       31976982 │
│ 183173490300605 │       12614988 │
│ 185609125465943 │       11621257 │
│ 189975612000676 │       17027391 │
│ 190487283562760 │       33010362 │
└─────────────────┴────────────────┘

```

**Comment by alexey-milovidov** [comment]
@nikitamikhaylov 
> This feature has unexpected results: (with big epsilon functions with anon_ prefix must be equal to ordinary function without prefix)

Or with small epsilon?

**Comment by nikitamikhaylov** [comment]
> Or with small epsilon?

 One naive approach is to add Laplace noise of scale 1/epsilon. And Laplace distribution with b as second parameter has variance equals to 2b^2. Other approaches have the same semantics. 

Also I will leave [paper](https://arxiv.org/pdf/1909.01917.pdf) here.

**Comment by alexey-milovidov** [comment]
This pull request is reviewed but the feature is found to be far from being practical
(there is no work on privacy budgets; guarantees of algorithms are unclear).

Close for now.

**Pull request closed without merging**